### PR TITLE
feat(MeetingsAdapter): add meeting stream states for local sharing

### PR DESCRIPTION
**Pull request by karinasigartau0798**
This enum store possible values for local share states.

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-246546

### CODE REVIEW

File: src/MeetingsAdapter.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ import WebexAdapter from './WebexAdapter';
  * @property {MediaStream}       remoteAudio  The remote audio stream
  * @property {MediaStream}       remoteShare  The remote media share stream
  * @property {MeetingState}      state        The state of the meeting
+ * @property {MeetingLocalShareState}      localShareState        The state of the local share
```
Comment (cipak) [inline src/MeetingsAdapter.js:32]: please re-align all meeting props with this one:
```js
/**
 * @property {MeetingState}           state            The state of the meeting
 * @property {MeetingLocalShareState} localShareState  The state of the local share
 * @property {TYPE}<one-ore-more-spaces>NAME<two-or-more-spaces>DESCRIPTION

File: src/MeetingsAdapter.js [hunk 2]
```diff
@@ -106,6 +107,19 @@ export const MediaPermissions = {
...
 };
 
+/**
+ * Enum for meeting local share state.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const MeetingLocalShareState = {
+  NOT_SHARING: 'NOT_SHARING', // set when the user has not yet started to share the screen
```
Comment (cipak) [inline src/MeetingsAdapter.js:117]: not sharing: local user is not sharing their screen
starting: the local sharing is in process of being started
sharing: local user is actively sharing their screen
stopping sharing: the local sharing is in process of being stopped

File: src/index.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ export {
   MeetingControlState,
   MeetingState,
   MediaPermissions,
+  MeetingLocalShareState,
```
Comment (cipak) [inline src/index.js:8]: keep it alphabetical

**Comment by cipak** [comment]
commit message: add local sharing states

File: src/MeetingsAdapter.js [hunk 4]
```diff
@@ -20,15 +20,16 @@ import WebexAdapter from './WebexAdapter';
...
+ * @property {string}                  ID               The meeting identifier
+ * @property {string}                  title            The title of the meeting
+ * @property {MediaStreamTrack}        localVideo       The local video stream track
+ * @property {MediaStreamTrack}        localAudio       The local audio stream track
+ * @property {MediaStream}             localShare       The local media share stream
+ * @property {MediaStream}             remoteVideo      The remote video stream
+ * @property {MediaStream}             remoteAudio      The remote audio stream
+ * @property {MediaStream}             remoteShare      The remote media share stream
+ * @property {MeetingState}            state            The state of the meeting
+ * @property {MeetingLocalShareState}  localShareState  The state of the local share
```
Comment (cipak) [inline src/MeetingsAdapter.js:32]: I think there should be a single space before `localShareState`, and two spaces after

File: src/MeetingsAdapter.js [hunk 5]
```diff
@@ -106,6 +107,19 @@ export const MediaPermissions = {
...
 };
 
+/**
+ * Enum for meeting local share state.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const MeetingLocalShareState = {
+  NOT_SHARING: 'NOT_SHARING', // local user is not sharing their screen
```
Comment (cipak) [inline src/MeetingsAdapter.js:117]: I think the standard was to have the values of enums as snake-case lowercase strings (eg: meeting control states, js sdk meeting event names), but we inadvertently changed that for media permissions and meeting states.

I would suggest using `'not-sharing'`, `'starting-sharing`, etc. for the state values.

File: src/MeetingsAdapter.js [hunk 6]
```diff
@@ -16,6 +16,14 @@ import WebexAdapter from './WebexAdapter';
  * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack}
  */
 
+/**
+ * An object that represents the local share information.
```
Comment (cipak) [inline src/MeetingsAdapter.js:20]: a media stream and its state

File: src/MeetingsAdapter.js [hunk 7]
```diff
@@ -16,6 +16,14 @@ import WebexAdapter from './WebexAdapter';
...
+ *
+ * @typedef {object} LocalShare
```
Comment (cipak) [inline src/MeetingsAdapter.js:22]: MediaStreamInfo

File: src/MeetingsAdapter.js [hunk 8]
```diff
@@ -106,6 +114,19 @@ export const MediaPermissions = {
   IGNORED: 'IGNORED', // set when the user has chosen to not interact with the device access
 };
 
+/**
+ * Enum for meeting local share state.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const MeetingLocalShareState = {
```
Comment (cipak) [inline src/MeetingsAdapter.js:123]: MeetingStreamState:
- ENABLED - media stream is actively sending data
- DISABLED - media stream is muted
- ENABLING
- DISABLING
- STOPPED - media stream has been stopped and can no longer be used
- NA - media stream cannot be obtained

**Comment by karinasigartau0798** [comment]
This PR was closed in favor of https://github.com/webex/component-adapter-interfaces/pull/69

**Pull request closed without merging**